app/components/emoji_picker/emoji_picker_base.js

```diff
@@ -187,12 +187,24 @@ export default class EmojiPicker extends PureComponent {
 
         clearTimeout(this.searchTermTimeout);
         const timeout = searchTerm ? 100 : 0;
-        this.searchTermTimeout = setTimeout(() => {
+        if (this.searchTermTimeout === undefined) {
             const filteredEmojis = this.searchEmojis(searchTerm);
             this.setState({
                 filteredEmojis,
             });
-        }, timeout);
+            this.searchTermTimeout = setTimeout(() => {
+                this.setState({
+                    filteredEmojis,
+                });
+            }, timeout);
```

Comment:
this is setting the same state again after 100ms, maybe what you can do is change the timeout to be something like

```js
const timeout = searchTerm && this.searchTimeout !== undefined ? 100 : 0;
```

Reply:
@enahum Instead, can we just give 0 for timeout in line 199?

Reply:
The idea is to remove the unnecessary setTimeout and setting the state twice

Reply:
So you mean instead of all these changes 
```js
const timeout = searchTerm && this.searchTimeout !== undefined ? 100 : 0;
```
This would fix this issue?

Reply:
I think it should

Reply:
@enahum  Even when the timeout is 0, React first would render and then the callback in the setTimeout would get called. So this would result in a jank showing `no emojis found` and then showing emojis.

Reply:
Because filteredEmojis is empty.
https://github.com/mattermost/mattermost-mobile/blob/master/app/components/emoji_picker/emoji_picker_base.js#L267

Reply:
@migbot sorry for the late reply. The filteredEmojis is empty because updating of filtered emojis is not synchronous. so on the initial render it would be an empty array and then setTimout callback is triggered which would update filtered emojis. fuse list is not empty.

Reply:
@migbot / @enahum, could one of you help @a-c-sreedhar-reddy with their question? Thanks!

Reply:
@migbot `searchTermTimout` will be used to determine if this was the first time a user typed in the search box.


If this is the first time then render the results synchronously ( if it is asynchronous then an empty screen would be rendered)
if this is not the first time then render the results once he stopped typing till then the old results would be displayed.


So the timeout function can be changed to a `no-op` function.
```
           this.searchTermTimeout = setTimeout(() => {
                //no-op
                return null;
            }, 0);
```

https://github.com/mattermost/mattermost-mobile/blob/master/app/components/emoji_picker/emoji_picker_base.js#L260 If we look here when the user starts typing for the first time `searchTerm` would not be empty but the `filtereEmojis` would be empty which would result in a jank of an empty screen and then the results.